### PR TITLE
Document how to launch kube-dns with Kubernetes on Mesos

### PR DESCRIPTION
**Pull request by sttts**
Fixes mesosphere/kubernetes-mesos#384

### CODE REVIEW

**Comment by F21** [comment]
Is the secrets volume still required? I started from scratch and built mesos binaries using the head of kubernetes.

After following the instructions, I noticed that there are no dns pods, rc or services after launching them:

```
root@mesos-master-01:~/kubernetes# kubectl get rc
CONTROLLER   CONTAINER(S)   IMAGE(S)   SELECTOR   REPLICAS
root@mesos-master-01:~/kubernetes# kubectl get services
NAME             LABELS                                    SELECTOR   IP(S)         PORT(S)
k8sm-scheduler   component=scheduler,provider=k8sm         <none>     10.10.10.21   10251/TCP
kubernetes       component=apiserver,provider=kubernetes   <none>     10.10.10.1    443/TCP
root@mesos-master-01:~/kubernetes# kubectl get pods
NAME      READY     STATUS    RESTARTS   AGE
```

**Comment by sttts** [comment]
With the described setup no secret is necessary anymore. 

Thanks for testing!

I will update the docs with the `--all-namespaces=true` parameter. Did this change in master tonight? I am pretty sure I saw the service yesterday in the default namespace.

**Comment by F21** [comment]
Awesome! Dealing with the secret volume was a bit troublesome, so I'm glad it's no longer required.

I think the original yaml files for launching kube-dns placed them under the `default` namespace. This was changed to `kube-system` in 90e105521c2ae08139e19e698d7acb5e61b30884 a few hours ago.

**Comment by karlkfi** [comment]
LGTM

P.S. I loath reference linking... So much harder to maintain than inline links. Does mungedocs even validate those?

File: docs/getting-started-guides/mesos.md [hunk 1]
```diff
@@ -191,6 +193,95 @@ Verify that the pod task is running in the Mesos web GUI. Click on the
...
+Then start the pod:
+
+```bash
+kubectl create -f busybox.yaml
+```
+
+When the pod is up and running, start a lookup:
+
+```bash
+kubectl  exec busybox -- nslookup kubernetes
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: maybe explain what this is looking up (the kubernetes master)

File: docs/getting-started-guides/mesos.md [hunk 2]
```diff
@@ -191,6 +193,95 @@ Verify that the pod task is running in the Mesos web GUI. Click on the
 Kubernetes framework. The next screen should show the running Mesos task that
 started the Kubernetes pod.
 
+## Launching kube-dns
+
+Kube-dns is an addon for Kubernetes which adds service discovery to the cluster. For a detailed explanation see [DNS in Kubernetes][4].
```
Comment (davidopp) [inline docs/getting-started-guides/mesos.md]: service discovery -> DNS-based service discovery (the env var based service discovery mechanism, while primitive, is available without DNS)

**Comment by davidopp** [comment]
looks good except for two minor nits

**Comment by sttts** [comment]
@davidopp addressed your two comments.

**Comment by davidopp** [comment]
LGTM

**Comment by erictune** [comment]
Seems like nothing in the added text is specific to Mesos.  If so, should the added text become its own document, linked here and in other places?

**Comment by karlkfi** [comment]
@erictune Feel free to make a PR to extract it after it's merged. ;)

**Pull request merged**